Tests Fail to Compile

`ninja check-clad` fails with error:

```
/clad/unittests/Kokkos/ParallelFor.cpp:114:3: error: user-defined derivative for 'parallel_for<Foo<Kokkos::View<double[5], Kokkos::HostSpace>>>' was provided but not used; expected signature 'void (const size_t, const Foo<Kokkos::View<double[5], Kokkos::HostSpace>> &, const size_t, const Foo<Kokkos::View<double[5], Kokkos::HostSpace>> &)' (aka 'void (const unsigned long, const Foo<Kokkos::View<double[5], Kokkos::HostSpace>> &, const unsigned long, const Foo<Kokkos::View<double[5], Kokkos::HostSpace>> &)') does not match
  114 |   Kokkos::parallel_for(5, f);
      |   ^
/clad/include/clad/Differentiator/KokkosBuiltins.h:515:6: note: candidate template ignored: could not match 'const ::std::string &' (aka 'const basic_string<char> &') against 'size_t' (aka 'unsigned long')
  515 | void parallel_for_pushforward(
      |      ^
/clad/include/clad/Differentiator/KokkosBuiltins.h:649:6: note: candidate template ignored: could not match 'const ::std::string &' (aka 'const basic_string<char> &') against 'size_t' (aka 'unsigned long')
  649 | void parallel_for_pushforward(
      |      ^
/clad/include/clad/Differentiator/KokkosBuiltins.h:709:6: note: candidate template ignored: could not match 'const ::std::string &' (aka 'const basic_string<char> &') against 'size_t' (aka 'unsigned long')
  709 | void parallel_for_pushforward(const ::std::string& str, const Policy& policy,
      |      ^
/clad/include/clad/Differentiator/KokkosBuiltins.h:721:6: note: candidate template ignored: could not match 'const Policy &' against 'size_t' (aka 'unsigned long')
  721 | void parallel_for_pushforward(const Policy& policy, const FunctorType& functor,
      |      ^
/clad/include/clad/Differentiator/KokkosBuiltins.h:728:6: note: candidate template ignored: could not match 'const Policy &' against 'size_t' (aka 'unsigned long')
  728 | void parallel_for_pushforward(
      |      ^
/clad/include/clad/Differentiator/KokkosBuiltins.h:738:6: note: candidate template ignored: could not match 'const ::std::string &' (aka 'const basic_string<char> &') against 'size_t' (aka 'unsigned long')
  738 | void parallel_for_pushforward(Policy policy, const FunctorType& functor,
      |      ^
/clad/include/clad/Differentiator/KokkosBuiltins.h:745:6: note: candidate template ignored: could not match 'const Policy &' against 'size_t' (aka 'unsigned long')
  745 | void parallel_for_pushforward(
      |      ^
In file included from /clad/unittests/Kokkos/ParallelFor.cpp:1:
In file included from /usr/include/Kokkos_Core.hpp:57:
In file included from /usr/include/KokkosCore_Config_DeclareBackend.hpp:48:
In file included from /usr/include/decl/Kokkos_Declare_OPENMP.hpp:49:
In file included from /usr/include/Kokkos_OpenMP.hpp:71:
```

--- a/include/clad/Differentiator/KokkosBuiltins.h
+++ b/include/clad/Differentiator/KokkosBuiltins.h
@@ -5,11 +5,12 @@
 #ifndef CLAD_DIFFERENTIATOR_KOKKOSBUILTINS_H
 #define CLAD_DIFFERENTIATOR_KOKKOSBUILTINS_H
 
+#include "clad/Differentiator/Differentiator.h"
 #include <Kokkos_Core.hpp>
+#include <Kokkos_ExecPolicy.hpp>
 #include <cstddef>
 #include <string>
 #include <type_traits>
-#include "clad/Differentiator/Differentiator.h"
 
 namespace clad::custom_derivatives {
 namespace class_functions {
@@ -733,6 +734,24 @@ void parallel_for_pushforward(
         ::Kokkos::is_execution_policy<Policy>::value>* /*d_param*/) {
   parallel_for_pushforward(::std::string("anonymous_parallel_for"), policy,
                            functor, ::std::string(""), d_policy, d_functor);
+}
+template <typename FunctorType>
+void parallel_for_pushforward(const size_t range, const FunctorType& functor,
+                              size_t d_range, const FunctorType& d_functor) {
+  parallel_for_pushforward(
+      ::std::string("anonymous_parallel_for"),
+      ::Kokkos::RangePolicy<size_t>(0, range), functor, ::std::string(""),
+      ::Kokkos::RangePolicy<size_t>(0, d_range), d_functor);
+}
+
+template <typename FunctorType>
+void parallel_for_pushforward(const ::std::string& str, size_t range,
+                              const FunctorType& functor,
+                              const ::std::string& d_str, size_t d_range,
+                              const FunctorType& d_functor) {
+  parallel_for_pushforward(
+      str, ::Kokkos::RangePolicy<size_t>(0, range), functor, d_str,
+      ::Kokkos::RangePolicy<size_t>(0, d_range), d_functor);
 }
 template <typename Policy, class FunctorType> // alternative signature
 void parallel_for_pushforward(Policy policy, const FunctorType& functor,

